005-nsinit-container-from-ipfs-filesystem/README.md

```diff
@@ -0,0 +1,98 @@
+# Starting a container with nsinit from an IPFS filesystem
+
+Docker's image format is built on the idea of [filesystem
+layers][layers].  You could conceivably create images with [a single
+tar layer][layer-changeset], but by packing the layer into a tar
+archive you lose many IPFS benefits.  For example, if you tweak a
+single file, you either have to distribute a new tar archive, or add a
+second tar layer that needs some fancy filesystem manipulations to
+unpack (e.g. a union- or snapshot-based filesystem like [AUFS][],
+[btrfs][], or [device-mapper's thin provisioning][thin-provisioning]).
+It's more efficient to store the filesystem directly in IPFS without
+any packing.
+
+The core of Docker's container launching is [libcontainer][], and
+libcontainer has [`nsinit`][nsinit], a demonstration utility that
+launches containers directly from a root filesystem.  `nsinit` reads
+the container's environment, networking, capabilities, etc. from a
+`container.json` in the root filesystem.  You can either put that file
+directly in your IPFS directory, or have users generate it themselves
+(you can get the standard configuration with `nsinit config`).
+
+The final wrinkle in this procedure is that [IPFS doesn't yet track
+file permissions][no-exec], so you can't launch `nsinit` directly from
+your IPNS directory.  As a workaround, the demo `Makefile` currently
+extracts the IPFS directory to your local filesystem, and then tweaks
+the file mode locally.
```

Comment:
@whyrusleeping maybe we can add perms to unixfs soon? we should consider it for sure in next week's sprint. https://github.com/ipfs/pm/issues/5